### DISCUSSION TITLE
Regrid wrong

### DISCUSSION THREAD

**Discussion by zhumingying** [dtcenter/METplus#2020] [Configuration] [answered]
I need to do precip verification of WRF fcst against STAGE4 precip data using METplus.
First I converter WRF fcst to NETCDF CF format; 
Second I regrid OBS of STAGE4 data using MET RegridDataPlane; and then do StatGrid of FSS. 

But the result is not right. It seems the Regrid obs Stage4 data is not right. see attached figures.
Could anyone help me? 

[test0.pdf](https://github.com/dtcenter/METplus/files/10486280/test0.pdf)
[test.pdf](https://github.com/dtcenter/METplus/files/10486277/test.pdf)
or see /glade/scratch/zhuming/virginia/test0.pdf and test.pdf

my configure files in cheyenne:
 /glade/work/zhuming/METplus4/RegridDataPlane_virginia_01h.conf
 /glade/work/zhuming/METplus4/GridStat_wrf_obsStgIV_GRIB_virginia.conf

**Comment by JohnHalleyGotway** [comment]
@zhumingying, I logged onto cheyenne to take a closer look.
I found your METplus log files in /glade/work/zhuming/data/metplus4/logs.
And in metplus.log.20230123190814, I see this command being run:
```
/glade/p/ral/jntp/MET/MET_releases/10.0.0/bin/regrid_data_plane \
-v 2 -method NEAREST -width 2 -field 'name="APCP"; level="L0";' -name APCP \
/glade/scratch/zhuming/virginia/st4_3h/ST4.2016062600.03h.nc \
"/glade/scratch/zhuming/virginia/cf/3dvar_conv/2016062300/wrfout_d01-cf.2016062300.nc" \
/glade/scratch/zhuming/virginia/st4_3h_regrid2/ST4.2016062600.03h.nc
```
And comparing that to the [usage statement](https://met.readthedocs.io/en/latest/Users_Guide/reformat_grid.html?highlight=regrid_data_plane#regrid-data-plane-usage) for regrid_data_plane, I see that:
1. `input_filename = /glade/scratch/zhuming/virginia/st4_3h/ST4.2016062600.03h.nc`
2. `to_grid = /glade/scratch/zhuming/virginia/cf/3dvar_conv/2016062300/wrfout_d01-cf.2016062300.nc`
3. `output_filename = /glade/scratch/zhuming/virginia/st4_3h_regrid2/ST4.2016062600.03h.nc`

So this is configured to regrid the Stage4 data to the grid defined by the wrfout file.

Is this what you WANT to do? Or do you intend to do the opposite, regridding the wrfout data to the StageIV domain?

**Comment by zhumingying** [comment]
Hi John,

Yes, I need to regrid the Stage4 data to the grid defined by the wrfout
file and then do the verification wrf fcst against stage4.

input file : /glade/scratch/zhuming/virginia/stage4/ST4.2016062312.01h
to grid = /glade/scratch/zhuming/virginia/cf/3dvar_conv/2016062300/
wrfout_d01-cf.2016062300.nc
output_file: /glade/scratch/zhuming/virginia/st4_1h_regrid2/
ST4.2016062312.01h.nc <http://st4.2016062600.03h.nc/>

Regrid config file: /glade/work/zhuming/METplus4/
RegridDataPlane_virginia_01h.conf


GridStat config file: /glade/work/zhuming/METplus4/
GridStat_wrf_obsStgIV_GRIB_virginia.conf

Redrid log file:
/glade/work/zhuming/data/metplus4/logs/metplus.log.20230124121605


GridStat log file:
/glade/work/zhuming/data/metplus4/logs/metplus.log.20230124123333


Both Regrid and GridStat METplus have successfully finished running, but it
looks regrid is not right, and GridStat results are not my expected.


See the figures in /glade/scratch/zhuming/virginia, test0.pdf (original)
and test.pdf (after regrid).


Thanks!

Zhuming

**Comment by JohnHalleyGotway** [comment]
@zhumingying, I ran the following commands with the MET plot_data_plane tool to see where MET thinks this gridded data exists.

1. Input Stage4 data:
```
plot_data_plane ST4.2016062600.03h.nc ST4.2016062600.03h.ps 'name="APCP";level="A1";'
```

<img width="500" alt="Screen Shot 2023-01-24 at 1 39 19 PM" src="https://user-images.githubusercontent.com/21087144/214405163-d514c835-1f83-4536-ae64-090d9dcfeb58.png">
2. The domain defined by the wrfout file:

```
plot_data_plane wrfout_d01-cf.2016062300.nc wrfout_d01-cf.2016062300.ps 'name="precip_1h"; level="(0,*,*)";'
```

<img width="498" alt="Screen Shot 2023-01-24 at 1 39 59 PM" src="https://user-images.githubusercontent.com/21087144/214405399-b6cc6325-3945-42be-abb3-301c1b920ab9.png">
3. The ST4 data regridded to that wrfout domain:

```
plot_data_plane ST4.2016062600.03h_wrfout_grid.nc ST4.2016062600.03h_wrfout_grid.ps 'name="APCP"; level="(*,*)";'
```

<img width="497" alt="Screen Shot 2023-01-24 at 1 40 56 PM" src="https://user-images.githubusercontent.com/21087144/214405732-6c965e81-248a-47cc-9194-e1bbcdb5cd21.png">

Comparing the plot in 1. to the plot in 3., the features seems to line up very nicely. I do not see any obvious problems with the result. However the plot in 3. DOES NOT match the image you've provided. It's possible that the problem exists not in the MET code but in the script you're using to plot the regridded StageIV data.

I'd recommend taking a look at that plotting script. And you can use `ncview` as a sanity check to make sure that the features of the data in the plot you're creating line up with the features from ncview.

**Reply by JohnHalleyGotway** [reply]
Sorry, I realize now that I should have used the specific files you indicated in your last post:
```
input file = /glade/scratch/zhuming/virginia/stage4/ST4.2016062312.01h
to_grid = /glade/scratch/zhuming/virginia/cf/3dvar_conv/2016062300/wrfout_d01-cf.2016062300.nc
output_file = /glade/scratch/zhuming/virginia/st4_1h_regrid2/ST4.2016062312.01h.nc
```

Plotting those (after renaming the 3rd)...
```
plot_data_plane ST4.2016062312.01h ST4.2016062312.01h.ps 'name="APCP";level="A1";'
plot_data_plane wrfout_d01-cf.2016062300.nc wrfout_d01-cf.2016062300.ps 'name="precip_1h"; level="(0,*,*)";'
plot_data_plane ST4.2016062312.01h_wrfout_regrid.nc ST4.2016062312.01h_wrfout_regrid.ps 'name="APCP";level="(*,*)";'
```

Regardless, the regridding results look fine to me...
FROM...
<img width="497" alt="Screen Shot 2023-01-24 at 1 59 48 PM" src="https://user-images.githubusercontent.com/21087144/214412486-e6645063-6171-4480-80a6-27f7c92e99e2.png">
TO...
<img width="498" alt="Screen Shot 2023-01-24 at 2 00 05 PM" src="https://user-images.githubusercontent.com/21087144/214412578-7eb11570-0f08-47dd-88a9-1aa8d9985417.png">

The features in the data before/after the regridding line up nicely.

**Comment by zhumingying** [comment]
Thanks!

So regrid is right.

Something is wrong in my GridStat? From the precip distribution of WRF fcst
(figure: rain_2016062321.pdf), It's for sure "CTRL" run is the worst, but
It has the best FSS within the first 3 hours, shown in figure of "
ss1h_2016062321.pdf".

/glade/scratch/zhuming/virginia/figure/rain_2016062321.pdf


/glade/scratch/zhuming/virginia/gridstat2/figure2/fss1h_2016062321.pdf


Some params are not set right in my GridStat config file? Would you check
it for me?


What "GRID_STAT_NEIGHBORHOOD_COV_THRESH" is for?


see config file /glade/work/zhuming/METplus4/
GridStat_wrf_obsStgIV_GRIB_virginia.conf

    log file:
/glade/work/zhuming/data/metplus4/logs/metplus.log.20230124123333

Thanks!
Zhuming

**Reply by JohnHalleyGotway** [reply]
Zhuming,

It sounds like you're asking me to figure out why one model outperforms other competing models when measured by Fractions Skill Score. That's definitely beyond the scope of the METplus Discussions support. I see the config file you referenced (/glade/work/zhuming/METplus4/GridStat_wrf_obsStgIV_GRIB_virginia.conf) and don't see any egregious problems with it.

METplus discussions is a good place for questions about runtime errors or further explaining configuration options. But we certainly aren't equipped to answer scientific questions about your data. The data analysis is up to you.

In general, the more specific and detailed you're able to make your question, the more we'll be able to help.

The GRID_STAT_NEIGHBORHOOD_COV_THRESH option in METplus sets the [nbrhd.cov_thresh](https://met.readthedocs.io/en/latest/Users_Guide/grid-stat.html?highlight=nbrhd#nbrhd) option in the MET config file. It defines the threshold applied to populate the NBRCTC and NBRCTS output line types. But FSS is in the NBRCNT line type. So you don't even need to define that since you don't need to compute NBRCTC and NBRCTS.

Thanks,
John

**Reply by JohnHalleyGotway** [reply]
Hi Zhuming, just wanted to follow up to see if you have any remaining questions about neighborhood methods in Grid-Stat? Or are we ready to mark this discussion as being answered?

**Comment by zhumingying** [comment, marked as answer]
The precip verification FSS results are not my expected, and I did not find
something/somewhere wrong in my processing.
So I have to use my old way to do the verification.

Thanks!
Zhuming
